Makefile

```diff
@@ -7,20 +7,32 @@ COVERAGE = python3-coverage
 
 DESTDIR=/
 
+.PHONY: all all-style all-fix \
+	build clean install \
+	check-all check check-tutorial check-nocompat check-compat \
+	check-pypy check-noextensions check-contrib \
+	typing style fix format reformat codespell \
+	coverage coverage-html doc sphinx apidocs
+
 all: build
 
-doc:: sphinx
+all-style:: style format codespell
 
-sphinx::
-	$(MAKE) -C docs html
+all-fix:: fix reformat codespell
```

Comment:
codespell doesn't make changes so it seems odd to be included here.

I'd also argue that reformat does something fundamentally different from "ruff check --fix" which does more than just change formatting - it can affect behaviour.

Reply:
That's news to me, and the [Ruff docs](https://docs.astral.sh/ruff/formatter/) make no mention of that, are you sure? The formatter is supposed to reformat the code only, not change how it parses. You might be thinking of `ruff check --unsafe-fixes`, which does say it could potentially change behaviour, but none of our rules use that.

Reply:
Ruff is not meant to make changes to behaviour but it's possible, especially in corner cases. Unlikely "ruff format" I'd be uncomfortable running it unconditionally, for example.

Reply:
I really think this would be considered a serious bug if `ruff format` changed any behaviour. IMHO splitting it out into separate rules that apply fixes, as opposed to simply checking, makes it pretty clear that that's their purpose. I don't think forcing people to apply all the mechanical formatting fixes manually to please the tool is an improvement, when the tool itself can apply them. This just extends the rules so that there's a comprehensive set of checks and matching fixes.

Reply:
I also added `codespell-fix` to match the rest. Since spelling fixes are not necessarily obvious, I made it ask for confirmation before applying each one.